crates/bevy_core_pipeline/src/skybox/skybox.wgsl

```diff
@@ -0,0 +1,51 @@
+#import bevy_render::view
+
+@group(0) @binding(0)
+var skybox: texture_cube<f32>;
+@group(0) @binding(1)
+var skybox_sampler: sampler;
+@group(0) @binding(2)
+var<uniform> view: View;
+
+struct VertexOutput {
+    @builtin(position) clip_position: vec4<f32>,
+    @location(0) world_position: vec3<f32>,
+};
+
+//  3 |  2.
+//  2 |  :  `.
+//  1 |  x-----x.
+//  0 |  |  s  |  `.
+// -1 |  0-----x.....1
+//    +---------------
+//      -1  0  1  2  3
+//
+// The axes are clip-space x and y. The region marked s is the visible region.
+// The digits in the corners of the right-angled triangle are the vertex
+// indices.
+@vertex
+fn skybox_vertex(@builtin(vertex_index) vertex_index: u32) -> VertexOutput {
+    // See the explanation above for how this works.
+    let clip_position = vec4(
+        f32(vertex_index & 1u),
+        f32((vertex_index >> 1u) & 1u),
+        0.25,
+        0.5
+    ) * 4.0 - vec4(1.0);
```

Comment:
```suggestion
    let clip_xy = vec2<f32>(f32(vertex_index & 1u), f32(vertex_index >> 1u)) * 4.0 - 1.0;
    let clip_position = vec4<f32>(clip_xy, 0.0, 1.0);
```

just to keep the logic closer to fullscreen.wgsl and reduce cognitive load a bit.

Reply:
I would agree but I remember it being argued by industry peeps that my version is faster.